lib/useKResponsiveWindow.js

```diff
@@ -0,0 +1,199 @@
+import { computed, onBeforeUnmount, onMounted, ref } from '@vue/composition-api';
+
+/**
+ * Returns window inner width and height
+ * @returns {Object} The window's inner width and height
+ */
+function windowMetrics() {
+  return {
+    width: window.innerWidth,
+    height: window.innerHeight,
+  };
+}
+
+/**
+ * Check if Nuxt is server side rendering
+ * @returns {Boolean}
+ */
+function isNuxtServerSideRendering() {
+  return process && process.server;
+}
+
+/**
+ * Add a media query
+ * @param {String} query
+ * @param {requestCallback} callback
+ * @returns {Object} The mediaQueryList, eventHandler, and stopListening
+ */
+function addMediaQuery(query, callback) {
+  if (isNuxtServerSideRendering()) {
+    return;
+  }
+
+  const mediaQueryList = window.matchMedia(query);
+  const eventHandler = event => {
+    callback(event);
+  };
+  const stopListening = () => {
+    mediaQueryList.removeEventListener('change', eventHandler);
+  };
+  mediaQueryList.addEventListener('change', eventHandler);
+
+  return {
+    mediaQueryList,
+    eventHandler,
+    stopListening,
+  };
+}
+
+/**
+ * Export window properties
+ * @returns {Object} Window properties
+ */
+export default function useKResponsiveWindow() {
+  /** Window properties */
+  const windowWidth = ref(null);
+  const windowHeight = ref(null);
+  const windowBreakpoint = ref(null);
+  const windowIsPortrait = ref(false);
+  const windowIsLandscape = ref(false);
+  const windowGutter = ref(16);
+  const windowIsShort = ref(false);
+  const windowIsLarge = computed(() => windowBreakpoint.value > 2);
+  const windowIsMedium = computed(() => windowBreakpoint.value === 2);
+  const windowIsSmall = computed(() => windowBreakpoint.value < 2);
+
+  /**
+   * @function addMediaQuery
+   */
+  const orientationQuery = addMediaQuery(`(orientation: portrait)`, event => {
+    updateOrientation(event.matches);
+    updateWindow();
+    updateGutter();
+  });
+
+  /**
+   * @function addMediaQuery
+   */
+  const heightQuery = addMediaQuery(`(max-height: 600px)`, event => {
+    windowIsShort.value = event.matches;
+    updateWindow();
+  });
+
+  /**
+   * Generate width media queries
+   */
+  const widthQueries = (() => {
+    const SCROLL_BAR = 16;
+    const queries = [
+      '(max-width: 480px)',
+      '(max-width: 600px)',
+      '(max-width: 840px)',
+      `(max-width: ${960 - SCROLL_BAR}px)`,
+      `(max-width: ${1280 - SCROLL_BAR}px)`,
+      `(max-width: ${1440 - SCROLL_BAR}px)`,
+      `(max-width: ${1600 - SCROLL_BAR}px)`,
+      `(min-width: 1601px)`,
+    ];
```

Comment:
The above 2 variables don't seem to change during calls to `useKResponsiveWindow` or `widthQueries`, so they could be moved further up in scope. Although I don't think it's a big deal.